rfcs/0000-changes-lifecyle.md

```diff
@@ -0,0 +1,256 @@
+- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [FuelLabs/sway-rfcs#0000](https://github.com/FuelLabs/sway-rfcs/pull/001)
+- Sway Issue: [FueLabs/sway#0000](https://github.com/FuelLabs/sway/issues/001)
+
+# Summary
+
+[summary]: #summary
+
+This RFC is a guide on how to introduce breaking changes following the compiler release lifecycle.
+
+# Motivation
+
+[motivation]: #motivation
+
+To maximize user experience when updating the compiler, we need a guide on how to introduce breaking changes.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+The compiler will follow a "rolling release" scheme, which means that periodically (to be specified) a
+new major version will be released.
+
+This means that as soon as the compiler reach version "1.0.0", the next **major version** will be "1.1.0"; and
+after that and if needed, the next **minor version** would be "1.1.1".
+
+The only changes that will trigger "minor updates" are:
+1 - Urgent security fixes;
+2 - Bugs that render "stable" functionality unusable;
+
+## Release Notes
+
+To track all "release notes" it will be create special file `ReleaseNotes.md` that will live in the repo root.
+
+When a version is being launched, for example `1.10` a new section `1.11-nightly` will be created empty. From this time on, all new PRs will create a new item under the "next version", `1.11` in this example.
+
+Each item will follow the template:
+
+```
+# Version 1.11-nightly
+
+## Forc
+
+### Breaking Changes
+### New Features
+### Bugs Fix
+### Others
+
+## Tools
+
+## Sway
+
+## `sway-lib-core` and `sway-lib-std`
+
+# Version 1.10
+
+...
+```
+
+Inside each section, items will have the following templae
+
+```
+- Index operator using Index trait [#6356](https://github.com/FuelLabs/sway/pull/6356)
+```
+
+Given that all PRs will touch this file, to avoid conflicts and decrease the experience when merging them, we will use "git custom merge driver" (see https://git-scm.com/docs/gitattributes#_defining_a_custom_merge_driver).
+
+This allows a custom merge strategy to a specific file using ".gitattributes".
+
+```
+ReleaseNotes.md merge=union
+```
+
+
+## Breaking changes
+
+A breaking change is a difference in functionality from the previous version of the compiler that may require an update to sway code in order for it to compile.
+
+The following changes are defined to be breaking changes, and will need to follow the process of being gated by features flags.
```

Comment:
Perhaps additional one: changing the way how `storage` field slots are calculated? Right now, we calculate them as `sha256("storage::ns1::ns2::..::nsN.field_name")` and want to introduce a new algorithm: `sha256((0u8, "storage::ns1::ns2::..::nsN.field_name"))`. There were questions on how slots/storage keys are calculated for `storage` fields, so I assume there are devs who rely on that.